lib/matplotlib/text.py

```diff
@@ -1222,9 +1222,7 @@ def set_fontproperties(self, fp):
         ----------
         fp : `.font_manager.FontProperties`
         """
-        if isinstance(fp, str):
-            fp = FontProperties(fp)
-        self._fontproperties = fp.copy()
+        self._fontproperties = FontProperties._from_any(fp).copy()
```

Comment:
Maybe `_from_any` should return copy always?

Reply:
See below